### ISSUE TITLE
-XX:+CMSClassUnloadingEnabled Breaks Service

### ISSUE THREAD

**Issue by JdeFalconr** [agowa338/MinecraftSystemdUnit#30]
This service fails to start if -XX:+CMSClassUnloadingEnabled is part of the Java arguments. I'm guessing this is related to the new Java 16 requirement but I don't really know much about Java. I figured this out (after using a completely different java command line to run this service - https://linuxconfig.org/how-to-setup-minecraft-server-on-ubuntu-18-04-bionic-beaver-linux) by selectively removing arguments until I figured out that particular one broke execution.

**Comment by BlenderVali** [comment]
This is for memory optimization, So if it breaks in a newer version of Java it is good to know about this. 
What version of Java are you using exactly?

**Comment by JdeFalconr** [comment]
These are the ones I have installed though I'm pretty confident it's the Java 16 ones that are at issue. BTW I'm running Ubuntu 20.04.2.

```
ii  openjdk-11-jre-headless:amd64        11.0.11+9-0ubuntu2~20.04          amd64        OpenJDK Java runtime, using Hotspot JIT (headless)
ii  openjdk-16-jdk-headless:amd64        16.0.1+9-1~20.04                  amd64        OpenJDK Development Kit (JDK) (headless)
ii  openjdk-16-jre-headless:amd64        16.0.1+9-1~20.04                  amd64        OpenJDK Java runtime, using Hotspot JIT (headless)
ii  openjdk-8-jre-headless:amd64         8u292-b10-0ubuntu1~20.04          amd64        OpenJDK Java runtime, using Hotspot JIT (headless)
```

**Comment by BlenderVali** [comment]
On https://wiki.openjdk.java.net/display/zgc/Main#Main-JDK16 it is stated:

_JDK 11
Initial version of ZGC
Does not support class unloading (using -XX:+ClassUnloading has no effect)_

_JDK 12
Support for concurrent class unloading
Further pause time reductions_

On https://docs.oracle.com/en/java/javase/11/tools/java.html is stated:

_Enables class unloading when using the concurrent **mark-sweep (CMS) garbage collector**. This option is **enabled** by default. **To disable class unloading** for the CMS garbage collector, specify -XX:-CMSClassUnloadingEnabled._

In conclusion the argument `-XX:+CMSClassUnloadingEnabled` seems to be for version 10 or lower as well as for the CMS garbage collector only.. At version 11 and above the ZGC garbage collector is used. So using this argument should result in a error message and process termination. 

Admittedly I myself don't know the reason why it is included in the first place. But I am sure a pull request would be welcome for future peeps.

**Comment by JdeFalconr** [comment]
I'm happy to help here but I'm not very experienced in Github. Can you point me to what I need to do?

**Comment by BlenderVali** [comment]
There probably need to be a lot of changes in order to make this repo jdk 16 compatible. I rather suggest, use jdk 8 or 11. Since this seem to be the supported versions. You seem to have 8 installed. Otherwise, check out https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request. Though admittedly it is going to be a lot of work that way (community driven projects and all.)

**Comment by JdeFalconr** [comment]
Thank you! I thought that Minecraft 1.17+ required Java 16 at a minimum, though? From what I understand anything less won't be an option.
________________________________

**Comment by agowa** [comment]
PR #31 has already been merged, so closing this ticket now.
